### PR TITLE
Adding the Circuit Dojo nRF9160 Feather device tree definitions.

### PR DESCRIPTION
**Pull request by jaredwolff**
This commit is for adding to the soon-to-be-released nRF9160 Feather. Let me know if there's anything else that needs to go in to make this happen!

### CODE REVIEW

**Comment by nzmichaelh** [comment]
LGTM

**Comment by jaredwolff** [comment]
Added onboard RTC to device tree definition.

**Comment by jaredwolff** [comment]
@nordic-krch @galak @MaureenHelm @nashif @jakub-uC let me know if you need any changes on this/questions/comments/concerns. Thank you! 😄

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
+	depends on BOARD_CIRCUITDOJO_FEATHER_NRF9160 && TRUSTED_EXECUTION_SECURE
+
+if BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS
+
+config FLASH_LOAD_OFFSET
+	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
+
+config FLASH_LOAD_SIZE
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
```
Comment (mbolivar-nordic) [inline boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig:35]: It looks like FLASH_LOAD_SIZE is always getting set to the size of `zephyr,code-partition`, regardless of whether it's a secure or non-secure configuration. The same default is set above.

Is there some potential cleanup to do here? If so I guess it should be done in a follow-up PR as this seems to be copy/pasted from the nRF9160DK Kconfig.defconfig. CC @anangl

File: boards/arm/circuitdojo_feather_nrf9160/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+     :alt: Circuit Dojo nRF9160 Feather
+
+     nRf9160 Feather (Credit: Circuit Dojo)
+
+Overview
+********
+
+The nRF9160 Feather by Circuit Dojo is a single-board development
+for bringing your LTE-M and NB-IoT applications to life. The circuitdojo_feather_nrf9160
+board configuration leverages the pre-existing support support for the Nordic Semiconductor
```
Comment (mbolivar-nordic) [inline boards/arm/circuitdojo_feather_nrf9160/doc/index.rst:18]:
```suggestion
board configuration leverages the pre-existing support for the Nordic Semiconductor
```

**Review by mbolivar-nordic — COMMENTED** [review]
A couple of minor notes from a quick skim. Seems nice overall!

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig (hunk 1 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig:35]: Hey @mbolivar-nordic,

You guys would know better than me. 😇 Don't the code partitions get resized based on the SPM in NCS?

Thanks for taking a look at this.

**Comment by jaredwolff** [comment]
Fixed that typo in the same commit.

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig (hunk 1 above)
Comment (mbolivar-nordic) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig:35]: > You guys would know better than me. innocent Don't the code partitions get resized based on the SPM in NCS?

Yep, it's a question for @anangl

Comment (anangl) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig:35]: It looks fine. See https://github.com/zephyrproject-rtos/zephyr/commit/0122bacd2cc4b01909f991667cb17d67e92d2364. `FLASH_LOAD_SIZE` won't get this default when the image is built without `TRUSTED_EXECUTION_SECURE` being set.

File: boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160_common.dts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,uart-mcumgr = &uart0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		blue_led: led_0 {
+			gpios = <&gpio0 3 GPIO_ACTIVE_LOW>;
+			label = "Green LED (D7)";
```
Comment (anangl) [inline boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160_common.dts:22]: Is it a blue LED or a green one?

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig (hunk 1 above)
Comment (ioannisg) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig:35]: @anangl is right, @mbolivar-nordic  :)

File: boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160_common.dts (hunk 3 above)
Comment (ioannisg) [reply boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160_common.dts:22]: @jaredwolff ^^

File: boards/arm/circuitdojo_feather_nrf9160/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,159 @@
+.. _circuitdojo_feather_nrf9160:
+
+nRf9160 Feather
```
Comment (ioannisg) [inline boards/arm/circuitdojo_feather_nrf9160/doc/index.rst:3]: Should this be nRF9160 Feather

Comment (ioannisg) [reply boards/arm/circuitdojo_feather_nrf9160/doc/index.rst:3]: Same applies to other occurrences below

**Review by ioannisg — CHANGES_REQUESTED** [review]
looks ok to me
I would like you to use NS instead of _NS in the Kconfig definitions for non-secure, see inline comment.
(this is the case for nRF boards with trustzone)

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.board [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# Copyright (c) 2018-2020 Nordic Semiconductor ASA
+# Copyright (c) 2020 Circuit Dojo LLC
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_NRF9160_SICA
+
+config BOARD_CIRCUITDOJO_FEATHER_NRF9160
+	bool "Circuit Dojo nRF9160 Feather"
+
+config BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS
```
Comment (ioannisg) [inline boards/arm/circuitdojo_feather_nrf9160/Kconfig.board:12]: Can you remove the underscore from "_NS" here?

Comment (ioannisg) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.board:12]: ...and in similar occurrences, if applicable

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig (hunk 1 above)
Comment (mbolivar-nordic) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig:35]: Sorry I was unclear. What I mean is something like this:

```
config FLASH_LOAD_OFFSET
	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_CODE_PARTITION)) if \
		BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS

config FLASH_LOAD_SIZE
	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION)) if \
		BOARD_CIRCUITDOJO_FEATHER_NRF9160_NS ||
		(BOARD_CIRCUITDOJO_FEATHER_NRF9160 && TRUSTED_EXECUTION_SECURE)
```

File: boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160_common.dts (hunk 3 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160_common.dts:22]: It's blue. I'll rename the label to be consistent.

File: boards/arm/circuitdojo_feather_nrf9160/doc/index.rst (hunk 4 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/doc/index.rst:3]: Will correct this.

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.board (hunk 5 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.board:12]: Can do!

**Comment by jaredwolff** [comment]
Addressed all concerns above:

Changed LED color to match (Blue in this case)
Removed underscore from NS suffix.
Fixed capitalization of nRF.

File: boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig (hunk 1 above)
Comment (ioannisg) [reply boards/arm/circuitdojo_feather_nrf9160/Kconfig.defconfig:35]: @mbolivar-nordic i fI read this correctly this is a "no-op" and looks correct to me, but i think it is a matter of taste at the end of the date. What worries me is that you might need to have the default setting in one loooong line.

**Review by ioannisg — COMMENTED** [review]
You still need to remove the underscore in _ns from the file names :)

**Comment by jaredwolff** [comment]
> You still need to remove the underscore in _ns from the file names :)

Oh yea. Duh. Pushed those files renamed without the _

**Review by ioannisg — APPROVED** [review]
LGTM now, thanks @jaredwolff 
@anangl pls take a look once more

**Comment by jaredwolff** [comment]
After some testing. Did find a need for a board.c which enables the power latch. Not sure if that should be smashed into one comment or separately since you guys already looked at the main commit.

File: boards/arm/circuitdojo_feather_nrf9160/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+* CMSIS-DAP based programmers
+
+Check out `Getting Started`_ for more info.
+
+Building an application
+=======================
+
+In most cases you'll want to use the ``_ns`` target with any of the Zephyr
+or Nordic based examples. Some of the examples do not use secure mode,
+so they do not required the ``_ns`` suffix. A great example of this is the
```
Comment (anangl) [inline boards/arm/circuitdojo_feather_nrf9160/doc/index.rst:100]: Now that the non-secure target is named `circuitdojo_feather_nrf9160ns`, this section should be updated.

File: boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160ns.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+identifier: circuitdojo_feather_nrf9160_ns
```
Comment (anangl) [inline boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160ns.yaml:1]: This should be also updated.

File: boards/arm/circuitdojo_feather_nrf9160/board.c [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+#define GPIO0 DT_LABEL(DT_NODELABEL(gpio0))
+#define POWER_LATCH_PIN 31
+
+static int board_circuitdojo_feather_nrf9160_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Get the device binding */
+	struct device *gpio = device_get_binding(DT_LABEL(DT_NODELABEL(gpio0)));
```
Comment (anangl) [inline boards/arm/circuitdojo_feather_nrf9160/board.c:18]: I guess you defined `GPIO0` with the intention to use it in this call.

File: boards/arm/circuitdojo_feather_nrf9160/board.c [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+	if (gpio == NULL) {
+		return -1;
```
Comment (anangl) [inline boards/arm/circuitdojo_feather_nrf9160/board.c:21]: You should rather use some errno code here. `-ENODEV` seems suitable.

File: boards/arm/circuitdojo_feather_nrf9160/board.c [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	int err;
+
+	/* Configure this pin as output. */
+	err = gpio_pin_configure(gpio, POWER_LATCH_PIN, GPIO_OUTPUT_ACTIVE);
```
Comment (anangl) [inline boards/arm/circuitdojo_feather_nrf9160/board.c:27]:
```suggestion
	err = gpio_pin_configure(gpio, POWER_LATCH_PIN, GPIO_OUTPUT_HIGH);
```
If you want to stay with `GPIO_OUTPUT_ACTIVE`, you should add also `GPIO_ACTIVE_HIGH`, as currently this code assumes that the value of this flag is `0`.

File: boards/arm/circuitdojo_feather_nrf9160/board.c [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+	if (err) {
+		return err;
+	}
+
+	/* Enable this pin to keep the board powered on battery. */
+	err = gpio_pin_set(gpio, POWER_LATCH_PIN, 1);
```
Comment (anangl) [inline boards/arm/circuitdojo_feather_nrf9160/board.c:33]: This is not needed. `gpio_pin_configure()` above sets the initial state of this pin.

**Comment by carlescufi** [comment]
@jaredwolff please address the outstanding review comments so we can merge this.

File: boards/arm/circuitdojo_feather_nrf9160/board.c (hunk 10 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/board.c]: Done!

File: boards/arm/circuitdojo_feather_nrf9160/board.c (hunk 11 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/board.c]: Done!

File: boards/arm/circuitdojo_feather_nrf9160/board.c (hunk 10 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/board.c]: Done!

File: boards/arm/circuitdojo_feather_nrf9160/board.c (hunk 9 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/board.c]: Fixed!

File: boards/arm/circuitdojo_feather_nrf9160/board.c (hunk 8 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/board.c]: Fixed!

File: boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160ns.yaml (hunk 7 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/circuitdojo_feather_nrf9160ns.yaml]: Fixed!

File: boards/arm/circuitdojo_feather_nrf9160/doc/index.rst (hunk 6 above)
Comment (jaredwolff) [reply boards/arm/circuitdojo_feather_nrf9160/doc/index.rst]: Fixed!

**Review by anangl — APPROVED** [review]

**Pull request merged**